test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -4889,6 +4889,278 @@ TEST_P(ExtProcIntegrationTest, SendHeaderBodyNotSendTrailerTest) {
   verifyDownstreamResponse(*response, 200);
 }
 
+TEST_P(ExtProcIntegrationTest, ServerWaitForBodyBeforeSendsHeaderRespMxn) {
+  config_helper_.setBufferLimits(1024, 1024);
```

Comment:
I haven't read through these tests but they are very long.

WDYT of factoring common stanzas into helper methods, so that each test reads like a much shorter list of calls to common helper functions, and it's easy to see the flow, and how tests differ?

Reply:
Good idea! done